AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -217,16 +217,18 @@ - (void)visibilityDidChange:(BOOL)isVisible
   [super visibilityDidChange:isVisible];
 
   if (_downloaderImplementsSetPriority) {
-    ASDN::MutexLocker l(_lock);
+    _lock.lock();
     if (_downloadIdentifier != nil) {
       if (isVisible) {
         [_downloader setPriority:ASImageDownloaderPriorityVisible withDownloadIdentifier:_downloadIdentifier];
       } else {
         [_downloader setPriority:ASImageDownloaderPriorityPreload withDownloadIdentifier:_downloadIdentifier];
       }
     }
+    _lock.unlock();
```

Comment:
@maicki Just so I understand what's going on, can you help correct me: It was my impression that the lock would unlock at this point automatically due to its destructor and it falling out of scope.